### PR TITLE
Major improvements+bugfixes

### PR DESCRIPTION
**Pull request by naslundx**
- Add graphics effects through the `FEffect` class and `FEffectType`.
- Fixed the occasional segfaults when getting hit by projectile
- Added `const` in many places where its applicable
- Added `getCenter()` to `FObject`
- Healthbar has background and turns red when there's little health left
- Powerups now give +100 health to characters and are properly removed

### CODE REVIEW

File: src/FEngine.cpp [hunk 1]
```diff
@@ -170,39 +185,50 @@ void FEngine::collisionDetection()
...
+	for (auto it = m_projectiles.begin(); m_projectiles.size() > 0 && it != m_projectiles.end();)
 	{
 		auto projectile = *it;
 		auto position = projectile.getPosition(), size = projectile.getSize();
 		bool outside = position.clampTo(0.f, 0.f, m_level->getWidth() * tileSize - size.x, m_level->getHeight() * tileSize - size.y);
 
 		if (outside || collisionLeft(projectile) || collisionRight(projectile) || collisionUp(projectile) || collisionDown(projectile))
 			m_projectiles.erase(it);
+		else
+			++it;
```
Comment (dekespo) [inline src/FEngine.cpp:214]: So these ++it's were the bug?

**Comment by dekespo** [comment]
good job indeed

File: src/FEngine.cpp (hunk 1 above)
Comment (naslundx) [inline src/FEngine.cpp:214]: Exactly.

**Comment by naslundx** [comment]
Thanks! I see now I forgot to merge

**Pull request merged**